crypto/cmp/cmp_util.c

```diff
@@ -106,129 +107,69 @@ const char *ossl_cmp_log_parse_metadata(const char *buf,
     return msg;
 }
 
-
-/*
- * auxiliary function for incrementally reporting texts via the error queue
- */
-static void put_error(int lib, const char *func, int reason,
-                      const char *file, int line)
+#define UNKNOWN_FUNC "(unknown function)" /* the default for OPENSSL_FUNC */
+static const char *improve_func_name(const char *func)
 {
-    ERR_new();
-    ERR_set_debug(file, line, func);
-    ERR_set_error(lib, reason, NULL /* no data here, so fmt is NULL */);
+    return func == NULL ? UNKNOWN_FUNC
+                        : strcmp(func, UNKNOWN_FUNC) == 0 ? "" : func;
```

Comment:
I wonder if you could clarify this with a comment somehow what this function is doing.
And are you sure this is the most simple way to achieve what you want to achieve here?

Reply:
I've added a little explaining comment.
The problem is that for C90 `__FUNCTION__`  is not officially defined, and `macros.h` then inserts the bulky but useless and slightly misleading text "(unknown function)". The function name would be known, but with pedantic compiler options it is not accessible. In such case I find an empty string more appropriate.

Reply:
I had experimented with various places and ways of replacing that "(unknown function)" string. 
Apart from modifying `include/openssl/macros.h`, (which would not be the topic of this PR) I had not found a simpler way of getting rid of the useless text output.